tests/data_io/test_core_connect.py

```diff
@@ -0,0 +1,49 @@
+"""Tests for related to the Tamr Core-connect"""
+import tamr_toolbox as tbox
+from tamr_toolbox import utils
+from tamr_toolbox.utils.testing import mock_api
+from tamr_unify_client import Client
+from tests._common import get_toolbox_root_dir
+
+CONFIG = utils.config.from_yaml(
+    get_toolbox_root_dir() / "tests/mocking/resources/toolbox_test.yaml"
+)
+
+DATASET_NAME = "test_core_connect"
+PRIMARY_KEYS = None
+
+enforce_online_test = False
+
+
+def remove_test_datasets(client: Client):
+    dataset_names = [
+        DATASET_NAME,
+        DATASET_NAME + "_2",
+        DATASET_NAME + "_profile",
+        DATASET_NAME + "_sample",
+        DATASET_NAME + "_2_sample",
+    ]
+    for dataset_name in dataset_names:
+        if tbox.dataset.manage.exists(client=client, dataset_name=dataset_name):
+            dataset = client.datasets.by_name(dataset_name)
+            dataset.delete()
+        assert not tbox.dataset.manage.exists(client=client, dataset_name=dataset_name)
+
+
+@mock_api(enforce_online_test=enforce_online_test)
+def test_ingest_dataset():
+    tamr_client = utils.client.create(**CONFIG["toolbox_test_instance"])
+    jdbc_connect = CONFIG["core_connect"]["jdbc"]["ingest"]
+
+    # Reset test datasets if they exist
+    # remove_test_datasets(client=tamr_client)
+
+    tbox.data_io.core_connect.ingest_dataset(
+        client=tamr_client,
+        jdbc_connect=jdbc_connect,
+        dataset_name=DATASET_NAME,
+        query="select * from dataset.dataset_ns_current limit 10",
+        primary_key=PRIMARY_KEYS,
+        # truncate_tamr_dataset="true",
+        retrieve_connect_metadata="true",
+    )
```

Comment:
You will want to do a little more here - an assert statement that the dataset now exists in Tamr for example. And potentially rerun the remove_test_datasets function.

Reply:
Assert statements are added: one assert statement for testing if an imported dataset exists; one assert statement for importing a dataset with pk and truncating the tamr dataset;